### PR TITLE
[WIP] introduce categories to exporterlist

### PR DESCRIPTION
**Pull request by cherti**
As @juliusv mentioned once in #prometheus, the list of external exporters has grown quite a bit. This attempts to increase overview by introducing categories for the third party exporters.

The chosen categories are a first proposal and might still be subject to change, depending on the feedback, any of which is greatly appreciated!

### CODE REVIEW

File: content/docs/instrumenting/exporters.md [hunk 1]
```diff
@@ -41,52 +41,71 @@ and maintained. We encourage the creation of more exporters but cannot
...
    * [Docker Hub exporter](https://github.com/infinityworksltd/docker-hub-exporter)
    * [GitHub exporter](https://github.com/infinityworksltd/github-exporter)
+   * [OpenWeatherMap exporter](https://github.com/RichiH/openweathermap_exporter)
+   * [Rancher exporter](https://github.com/infinityworksltd/prometheus-rancher-exporter)
+
+### Logging
    * [Google's mtail log data extractor](https://github.com/google/mtail)
    * [Grok exporter](https://github.com/fstab/grok_exporter)
+   
+### Metrics-bridges
```
Comment (brian-brazil) [inline content/docs/instrumenting/exporters.md]: These are exporters, not bridges

"Monitoring systems" would be better

File: content/docs/instrumenting/exporters.md [hunk 2]
```diff
@@ -41,52 +41,71 @@ and maintained. We encourage the creation of more exporters but cannot
...
+
+### Services
+   * [BIG-IP exporter](https://github.com/ExpressenAB/bigip_exporter)
+   * [BIND exporter](https://github.com/digitalocean/bind_exporter)
+   * [PowerDNS exporter](https://github.com/janeczku/powerdns_exporter)
+   * [SMTP/Maildir MDA blackbox prober](https://github.com/cherti/mailexporter)
+
+### Miscellaneous
+   * [Django exporter](https://github.com/korfuri/django-prometheus)
    * [Heka dashboard exporter](https://github.com/docker-infra/heka_exporter)
```
Comment (brian-brazil) [inline content/docs/instrumenting/exporters.md]: Heka is a monitoring system

File: content/docs/instrumenting/exporters.md [hunk 3]
```diff
@@ -41,52 +41,71 @@ and maintained. We encourage the creation of more exporters but cannot
...
-   * [IPMI exporter](https://github.com/lovoo/ipmi_exporter)
    * [Jenkins exporter](https://github.com/lovoo/jenkins_exporter)
-   * [knxd exporter](https://github.com/RichiH/knxd_exporter)
    * [Meteor JS web framework exporter](https://atmospherejs.com/sevki/prometheus-exporter)
    * [Minecraft exporter module](https://github.com/Baughn/PrometheusIntegration)
-   * [Mirth Connect exporter](https://github.com/vynca/mirth_exporter)
-   * [MongoDB exporter](https://github.com/dcu/mongodb_exporter)
-   * [Munin exporter](https://github.com/pvdh/munin_exporter)
-   * [NATS exporter](https://github.com/lovoo/nats_exporter)
    * [New Relic exporter](https://github.com/jfindley/newrelic_exporter)
```
Comment (brian-brazil) [inline content/docs/instrumenting/exporters.md]: Monitoring system

File: content/docs/instrumenting/exporters.md [hunk 4]
```diff
@@ -41,52 +41,71 @@ and maintained. We encourage the creation of more exporters but cannot
...
-   * [ScaleIO exporter](https://github.com/syepes/sio2prom)
-   * [scollector exporter](https://github.com/tgulacsi/prometheus_scollector)
-   * [SMTP/Maildir MDA blackbox prober](https://github.com/cherti/mailexporter)
    * [Speedtest.net exporter](https://github.com/RichiH/speedtest_exporter)
-   * [SQL query result set metrics exporter](https://github.com/chop-dbhi/prometheus-sql)
-   * [Ubiquiti UniFi exporter](https://github.com/mdlayher/unifi_exporter)
-   * [Varnish exporter](https://github.com/jonnenauha/prometheus_varnish_exporter)
    * [WebDriver exporter](https://github.com/mattbostock/webdriver_exporter)
    * [Xen exporter](https://github.com/lovoo/xenstats_exporter)
    * [Zookeeper exporter](https://github.com/carlpett/zookeeper_exporter)
```
Comment (brian-brazil) [inline content/docs/instrumenting/exporters.md]: Database

File: content/docs/instrumenting/exporters.md [hunk 5]
```diff
@@ -41,52 +41,71 @@ and maintained. We encourage the creation of more exporters but cannot
...
-   * [RabbitMQ exporter](https://github.com/kbudde/rabbitmq_exporter)
-   * [RabbitMQ Management Plugin exporter](https://github.com/deadtrickster/prometheus_rabbitmq_exporter)
-   * [Rancher exporter](https://github.com/infinityworksltd/prometheus-rancher-exporter)
-   * [Redis exporter](https://github.com/oliver006/redis_exporter)
-   * [RethinkDB exporter](https://github.com/oliver006/rethinkdb_exporter)
    * [rTorrent exporter](https://github.com/mdlayher/rtorrent_exporter)
-   * [ScaleIO exporter](https://github.com/syepes/sio2prom)
-   * [scollector exporter](https://github.com/tgulacsi/prometheus_scollector)
-   * [SMTP/Maildir MDA blackbox prober](https://github.com/cherti/mailexporter)
    * [Speedtest.net exporter](https://github.com/RichiH/speedtest_exporter)
```
Comment (brian-brazil) [inline content/docs/instrumenting/exporters.md]: API?

File: content/docs/instrumenting/exporters.md [hunk 6]
```diff
@@ -41,52 +41,71 @@ and maintained. We encourage the creation of more exporters but cannot
...
+### Logging
    * [Google's mtail log data extractor](https://github.com/google/mtail)
    * [Grok exporter](https://github.com/fstab/grok_exporter)
+   
+### Metrics-bridges
+   * [Cloud Foundry Firehose exporter](https://github.com/cloudfoundry-community/firehose_exporter)
+   * [scollector exporter](https://github.com/tgulacsi/prometheus_scollector)
+   * [Munin exporter](https://github.com/pvdh/munin_exporter)
+
+### Services
```
Comment (brian-brazil) [inline content/docs/instrumenting/exporters.md]: Network services?

File: content/docs/instrumenting/exporters.md [hunk 7]
```diff
@@ -41,52 +41,71 @@ and maintained. We encourage the creation of more exporters but cannot
...
+   * [BIG-IP exporter](https://github.com/ExpressenAB/bigip_exporter)
+   * [BIND exporter](https://github.com/digitalocean/bind_exporter)
+   * [PowerDNS exporter](https://github.com/janeczku/powerdns_exporter)
+   * [SMTP/Maildir MDA blackbox prober](https://github.com/cherti/mailexporter)
+
+### Miscellaneous
+   * [Django exporter](https://github.com/korfuri/django-prometheus)
```
Comment (brian-brazil) [inline content/docs/instrumenting/exporters.md]: This isn't an exporter, it falls under utilities

File: content/docs/instrumenting/exporters.md [hunk 8]
```diff
@@ -41,52 +41,71 @@ and maintained. We encourage the creation of more exporters but cannot
...
+   * [NSQ exporter](https://github.com/lovoo/nsq_exporter)
+   * [RabbitMQ exporter](https://github.com/kbudde/rabbitmq_exporter)
+   * [RabbitMQ Management Plugin exporter](https://github.com/deadtrickster/prometheus_rabbitmq_exporter)
+   * [Mirth Connect exporter](https://github.com/vynca/mirth_exporter)
+
+### Storage
+   * [Ceph exporter](https://github.com/digitalocean/ceph_exporter)
+   * [ScaleIO exporter](https://github.com/syepes/sio2prom)
+
+### Webserver- and proxies
```
Comment (brian-brazil) [inline content/docs/instrumenting/exporters.md]: HTTP?

File: content/docs/instrumenting/exporters.md [hunk 9]
```diff
@@ -41,52 +41,71 @@ and maintained. We encourage the creation of more exporters but cannot
...
-   * [RethinkDB exporter](https://github.com/oliver006/rethinkdb_exporter)
    * [rTorrent exporter](https://github.com/mdlayher/rtorrent_exporter)
-   * [ScaleIO exporter](https://github.com/syepes/sio2prom)
-   * [scollector exporter](https://github.com/tgulacsi/prometheus_scollector)
-   * [SMTP/Maildir MDA blackbox prober](https://github.com/cherti/mailexporter)
    * [Speedtest.net exporter](https://github.com/RichiH/speedtest_exporter)
-   * [SQL query result set metrics exporter](https://github.com/chop-dbhi/prometheus-sql)
-   * [Ubiquiti UniFi exporter](https://github.com/mdlayher/unifi_exporter)
-   * [Varnish exporter](https://github.com/jonnenauha/prometheus_varnish_exporter)
    * [WebDriver exporter](https://github.com/mattbostock/webdriver_exporter)
```
Comment (brian-brazil) [inline content/docs/instrumenting/exporters.md]: This would fall under http

**Comment by juliusv** [comment]
If we do this, it seems odd to only categorize third-party exporters, but leave the official ones uncategorized. Repeating all the categories in both sections also does not seem great. Should we just join the official+third-party ones into one big section, but mark the official ones somehow?

File: content/docs/instrumenting/exporters.md (hunk 9 above)
Comment (cherti) [inline content/docs/instrumenting/exporters.md]: I thougt about that but wondered if I should merge the Server-side with the Client-side. But renaming the category to HTTP would solve that indeed

**Comment by cherti** [comment]
Indeed. Actually the official ones are comparatively few ones so introducing categories there seems a little over the top anyways.

But when we go for categories, it might indeed make sense to merge so that the page is more categorie-oriented that officiall/thirdparty-oriented.

File: content/docs/instrumenting/exporters.md (hunk 8 above)
Comment (cherti) [inline content/docs/instrumenting/exporters.md]: You mean `HTTP` as the Category name? Or `HTTP-Webservers...`-something. Just to make sure we don't have misunderstandings here and reiterate this one. :)

**Comment by juliusv** [comment]
Yep. Maybe just put `**(official)**` behind the name of the exporter? But keep+adjust the existing explanations around officially supported vs. third-party exporters at the top.

File: content/docs/instrumenting/exporters.md (hunk 8 above)
Comment (brian-brazil) [inline content/docs/instrumenting/exporters.md]: HTTP is the category name

File: content/docs/instrumenting/exporters.md (hunk 1 above)
Comment (cherti) [inline content/docs/instrumenting/exporters.md]: how about "other monitoring systems"? sounds better to me in the context of this, but that just a feeling.

Comment (juliusv) [inline content/docs/instrumenting/exporters.md]: "Legacy monitoring systems". Just kidding :P

File: content/docs/instrumenting/exporters.md (hunk 6 above)
Comment (cherti) [inline content/docs/instrumenting/exporters.md]: If mail is a network service then HTTP is also a network service. Network services is an improvement, but maybe the category itself is not a so good idea.

File: content/docs/instrumenting/exporters.md (hunk 1 above)
Comment (cherti) [inline content/docs/instrumenting/exporters.md]: hm... I can do it...^^

File: content/docs/instrumenting/exporters.md (hunk 7 above)
Comment (cherti) [inline content/docs/instrumenting/exporters.md]: Introducing a new category on the page, you mean?

**Comment by cherti** [comment]
@brian-brazil 
your sorting proposals should all be implemented now. feel free to go over it to check.

I was wondering for things like the jenkins-exporter and the minecraft-exporter if we want something like a `Applications`-category. Except that `Applications` would then include half of the exporters there. But something for like high-level-non-infrastructure-software or so?

**Comment by cherti** [comment]
@juliusv 
We could do something like that, but I'd leave out the stars. (or did you mean the markdown-syntax there?)

File: content/docs/instrumenting/exporters.md [hunk 10]
```diff
@@ -41,55 +41,75 @@ and maintained. We encourage the creation of more exporters but cannot
...
-   * [Rancher exporter](https://github.com/infinityworksltd/prometheus-rancher-exporter)
    * [Redis exporter](https://github.com/oliver006/redis_exporter)
    * [RethinkDB exporter](https://github.com/oliver006/rethinkdb_exporter)
-   * [rTorrent exporter](https://github.com/mdlayher/rtorrent_exporter)
-   * [ScaleIO exporter](https://github.com/syepes/sio2prom)
-   * [scollector exporter](https://github.com/tgulacsi/prometheus_scollector)
-   * [SMTP/Maildir MDA blackbox prober](https://github.com/cherti/mailexporter)
-   * [Speedtest.net exporter](https://github.com/RichiH/speedtest_exporter)
    * [SQL query result set metrics exporter](https://github.com/chop-dbhi/prometheus-sql)
+   * [Zookeeper exporter](https://github.com/carlpett/zookeeper_exporter)
```
Comment (brian-brazil) [inline content/docs/instrumenting/exporters.md]: Incidentally, we should be pointing people at the JMX exporter for this instead.

File: content/docs/instrumenting/exporters.md [hunk 11]
```diff
@@ -41,55 +41,75 @@ and maintained. We encourage the creation of more exporters but cannot
...
    * [Varnish exporter](https://github.com/jonnenauha/prometheus_varnish_exporter)
    * [WebDriver exporter](https://github.com/mattbostock/webdriver_exporter)
+   
+### APIs
+   * [Docker Hub exporter](https://github.com/infinityworksltd/docker-hub-exporter)
+   * [GitHub exporter](https://github.com/infinityworksltd/github-exporter)
+   * [OpenWeatherMap exporter](https://github.com/RichiH/openweathermap_exporter)
+   * [Rancher exporter](https://github.com/infinityworksltd/prometheus-rancher-exporter)
+   * [Speedtest.net exporter](https://github.com/RichiH/speedtest_exporter)
+   * [WebDriver exporter](https://github.com/mattbostock/webdriver_exporter)
```
Comment (brian-brazil) [inline content/docs/instrumenting/exporters.md]: This doesn't fit here

File: content/docs/instrumenting/exporters.md [hunk 12]
```diff
@@ -41,55 +41,75 @@ and maintained. We encourage the creation of more exporters but cannot
...
+
+### Logging
+   * [Google's mtail log data extractor](https://github.com/google/mtail)
+   * [Grok exporter](https://github.com/fstab/grok_exporter)
+   
+### other Monitoring Systems
```
Comment (brian-brazil) [inline content/docs/instrumenting/exporters.md]: Capital letter

File: content/docs/instrumenting/exporters.md (hunk 7 above)
Comment (brian-brazil) [inline content/docs/instrumenting/exporters.md]: There's a section down the bottom for them already

File: content/docs/instrumenting/exporters.md (hunk 11 above)
Comment (cherti) [inline content/docs/instrumenting/exporters.md]: oops, that shouldn't have gone the anyways

File: content/docs/instrumenting/exporters.md (hunk 10 above)
Comment (cherti) [inline content/docs/instrumenting/exporters.md]: Still, although this exporter seems to use a specific extraction feature of zookeeper?

Comment (brian-brazil) [inline content/docs/instrumenting/exporters.md]: It does some odd things that are non-recommended for an exporter. JMX will provide more typical stats.

Comment (cherti) [inline content/docs/instrumenting/exporters.md]: so you say dropping it entirely?

Comment (brian-brazil) [inline content/docs/instrumenting/exporters.md]: Probably the right thing to do. It's for a very specialised use case.

**Comment by cherti** [comment]
ok, I think we converged onto this one.

Questions remaining:
* any reordering of the categories on the site?
* doing the merge of official exporters we discussed directly in here or in a separate PR?

File: content/docs/instrumenting/exporters.md [hunk 13]
```diff
@@ -41,55 +41,72 @@ and maintained. We encourage the creation of more exporters but cannot
...
-   * [Speedtest.net exporter](https://github.com/RichiH/speedtest_exporter)
    * [SQL query result set metrics exporter](https://github.com/chop-dbhi/prometheus-sql)
+
+### Hardware-related
+   * [IoT Edison exporter](https://github.com/roman-vynar/edison_exporter)
+   * [IPMI exporter](https://github.com/lovoo/ipmi_exporter)
+   * [knxd exporter](https://github.com/RichiH/knxd_exporter)
    * [Ubiquiti UniFi exporter](https://github.com/mdlayher/unifi_exporter)
+
+### Messaging-Systems
```
Comment (juliusv) [inline content/docs/instrumenting/exporters.md]: There shouldn't be a hyphen here.

We don't use title case in docs headlines, so only the first word in headings should be capitalized.

**Comment by juliusv** [comment]
@cherti Feeling ok either way on doing the official/third-party merge either now or later. Fine to keep it for a later PR.

**Comment by cherti** [comment]
Then I'd say merge this one once we converged on what's left for the third-party-exporters. Then we can do the merge of all exporters with a fresh diff, which should be better readable for review.

File: content/docs/instrumenting/exporters.md (hunk 13 above)
Comment (juliusv) [inline content/docs/instrumenting/exporters.md]: Correct.

Comment (cherti) [inline content/docs/instrumenting/exporters.md]: Done

**Comment by cherti** [comment]
@brian-brazil @juliusv 
I'm not so convinced about the `Network services`-section. If mail is a network service, HTTP is also a network service. I'd propose we either get rid of that section entirely and put those services under `misc`, or we find a better categorization scheme, ideally including some more things from the `misc`-section. But I'd be fine with simply dropping the `network services` to `misc`.

**Comment by cherti** [comment]
@brian-brazil @juliusv 
I have now dropped the section `Network services` as it overlapped too much with other categories and sorted the things left in there into `miscellaneous`.

If there are no further issues this should be ready for merge.

**Comment by juliusv** [comment]
@brian-brazil will have stronger opinions than me, but I'll merge this for now and anyone who wants to change things can still do so in another PR :)

Thanks!

**Pull request merged**